addons/mail/static/src/discuss/core/common/store_service_patch.js

```diff
@@ -63,7 +63,7 @@ const storeServicePatch = {
         return Object.values(this.Thread.records)
             .filter((thread) => thread.channel_type === "chat" && thread.correspondent)
             .sort((a, b) => compareDatetime(b.lastInterestDt, a.lastInterestDt) || b.id - a.id)
-            .map((thread) => thread.correspondent.persona.id);
+            .map((thread) => thread.correspondent.partner_id.id);
```

Comment:
This was actually wrong before this PR if the model wasn't checked/filtered. This should be done as a bug fix in the first version with the issue, as it would incorrectly add guests in here.

Reply:
Well I guess it was implicitly checked through `thread.channel_type === "chat"` and the fact we normally don't create chat with guests, so there is no actual bug.